server/src/storage/types.rs

```diff
@@ -150,7 +132,279 @@ impl SqlType {
         }
         v
     }
+    /// compare function that lets you logical compare slices of u8
+    /// returns a boolean on success and Error on fail
+    /// uses other compare fn for the actual compare
+    pub fn cmp(&self, val: &[u8], val2: &[u8], comp: CompType)
+    -> Result<bool, Error>
+    {
+        info!("checking Compare type: {:?}", comp);
+        match self {
+            &SqlType::Int => {
+                match comp {
+                    CompType::Equ => {
+                        self.equal_for_int_with_value(val, val2)
+                    },
+                    CompType::NEqu => {
+                        self.equal_for_int_with_value(val, val2).map(|x| !x)
+                    },
+                    CompType::GThan => {
+                        self.greater_than_for_int_with_value(val, val2)
+                    },
+                    CompType::SThan => {
+                        self.lesser_than_for_int_with_value(val, val2)
+                    },
+                    CompType::GEThan => {
+                        self.lesser_than_for_int_with_value(val, val2).map(|x| !x)
+                    },
+                    CompType::SEThan => {
+                        self.greater_than_for_int_with_value(val, val2).map(|x| !x)
+                    },
+                }
+            },
 
+            &SqlType::Bool => {
+                match comp {
+                    CompType::Equ => {
+                        self.compare_as_bool(val, val2)
+                    },
+                    CompType::NEqu => {
+                        self.compare_byte_for_equal(val, val2).map(|x| !x)
+                    },
+                    CompType::GThan => {
+                        self.compare_byte_greater_than(val, val2) },
+                    CompType::SThan => {
+                        self.compare_byte_lesser_than(val, val2)
+                    },
+                    CompType::GEThan => {
+                        self.compare_byte_lesser_than(val, val2).map(|x| !x)
+                    },
+                    CompType::SEThan => {
+                        self.compare_byte_greater_than(val, val2).map(|x| !x)
+                    },
```

Comment:
`> >= < <=` comparisons for bool are pretty useless... We would just disallow those queries.